src/services/has-many-getter.js

```diff
@@ -80,26 +80,38 @@ function HasManyGetter(model, association, opts, params) {
         var query = association.find(conditions);
         handlePopulate(query);
 
-        return query;
+        return query.then(function(records) {
+          return [records, recordIds];
+        });
       })
-      .then(function(records) {
-        if (params.sort) {
-          var fieldSort = params.sort;
-          var descending = false;
-
-          if (params.sort[0] === '-') {
-            fieldSort = params.sort.substring(1);
-            descending = true;
-          }
+      .then(function(recordsAndRecordIds) {
+        var records = recordsAndRecordIds[0];
+        var fieldSort = params.sort;
+        var descending = false;
+
+        if (params.sort && (params.sort[0] === '-')) {
+          fieldSort = params.sort.substring(1);
+          descending = true;
+        }
 
+        if (fieldSort && (fieldSort !== 'parentsReferencingOrder')) {
           var recordsSorted = _.sortBy(records, function(record) {
             return record[fieldSort];
           });
 
           return descending ? recordsSorted.reverse() : recordsSorted;
         }
-
-        return records;
+        // console.log(`params.sort is not defined: ${JSON.stringify(params.sort)} for records: ${records.map(r => r._id)}`);
+        var recordIds = recordsAndRecordIds[1];
+        var recordIdStrings = recordIds.map(function(recordId) {
+          // Convert values to strings, so ObjectIds could be easily searched and compared.
+          return '' + recordId;
```

Comment:
Can't you "cast" the `record._id` to a String instead of the concatenation with an empty String for a cleaner code.

Reply:
accepted - using `String(...)` for casting